### PR TITLE
Fix xi-mac command line being able to open folders

### LINKED ISSUE #424: Can't open files after accidentally trying to open a directory
What it says on the tin. Trying to open a directory using the command line causes xi-mac to activate but seemingly do nothing. Additionally further attempts to open actual files via the command line or the GUI seem to do nothing at all.

### PR DESCRIPTION
**Pull request by nangtrongvuon**
## Summary

Small fix to make xi-mac CLI not open folders.

Should close #424.

## Review Checklist

- [x] I have responded to reviews and made changes where appropriate.
- [x] I have tested the code
- [x] I have updated comments / documentation related to the changes I made.
- [x] I have rebased my PR branch onto xi-mac/master.

### CODE REVIEW

**Comment by mmatoszko** [comment]
It seems that the reason it crashes on CI is referencing [xi-editor](https://travis-ci.com/xi-editor/xi-mac/builds/101312245) submodule commit which is not available yet. Do we need to update `xi-editor` in this PR?

**Comment by nangtrongvuon** [comment]
Oh, I think it's actually pointing to my quick open branch on xi-editor. I'll amend my commit with the proper master.

File: Sources/XiCliCore/CommandLineTool.swift [hunk 1]
```diff
@@ -70,6 +80,10 @@ public final class CommandLineTool {
         }
         
         let pathString = filePath.path
+
+        guard !pathIsDirectory(pathString) else {
```
Comment (mmatoszko) [inline Sources/XiCliCore/CommandLineTool.swift]: nit: Wouldn't it be more readable if we didn't inverse that logic?
What do you think of something like:
```
if pathIsDirectory(pathString) else {
    throw CliError.pathIsDirectory
}
```
? 😄

Comment (nangtrongvuon) [reply Sources/XiCliCore/CommandLineTool.swift]: Doesn't this mean throwing if the path is not a directory? Shouldn't this be 
`if pathIsDirectory(pathString) {
    throw CliError.pathIsDirectory
}`
?

Comment (mmatoszko) [reply Sources/XiCliCore/CommandLineTool.swift]: You’re right, my code won’t even compile 😅

Comment (nangtrongvuon) [reply Sources/XiCliCore/CommandLineTool.swift]: :) Though I think your comment makes sense, as not inverting `guard` logic seems to make it easier to read. I'll change this up a bit.

File: Sources/XiCliCore/CommandLineTool.swift [hunk 2]
```diff
@@ -52,6 +52,16 @@ public final class CommandLineTool {
...
         let fileManager = FileManager.default
         var filePath: URL!
+
+        // Small helper function used to determine if path is not a folder.
+        func pathIsNotDirectory(_ path: String) -> Bool {
+            var isDirectory = ObjCBool(false)
+            if fileManager.fileExists(atPath: path, isDirectory: &isDirectory) {
+                return !isDirectory.boolValue
+            } else {
+                return false
```
Comment (mmatoszko) [inline Sources/XiCliCore/CommandLineTool.swift]: I think we can now return `true` here :)

Comment (nangtrongvuon) [reply Sources/XiCliCore/CommandLineTool.swift]: You're right, going to fix that :)

**Comment by mmatoszko** [comment]
:rocket:

**Pull request merged**